### PR TITLE
libssh2: update to version 1.9.0 (security fix)

### PR DESCRIPTION
**Pull request by ja-pa**
Maintainer: @jslachta 
Compile tested: Turris Omnia (TOS5), OpenWrt master
Run tested: N/A

Description:
Fix CVE-2019-13115 - https://nvd.nist.gov/vuln/detail/CVE-2019-13115
Changelog https://www.libssh2.org/changes.html
Remove old patches
Signed-off-by: Jan Pavlinec <jan.pavlinec@nic.cz>

### CODE REVIEW

**Comment by diizzyy** [comment]
Can we convert it to CMake while at it?

**Comment by ja-pa** [comment]
@diizzyy Just update PR with CMake version

**Comment by neheb** [comment]
Please rebase against current master to fix the buildbot.

File: libs/libssh2/Makefile [hunk 1]
```diff
@@ -46,12 +47,18 @@ define Package/libssh2/config
...
-	$(if $(CONFIG_LIBSSH2_OPENSSL),--with-openssl --with-libssl-prefix=$(STAGING_DIR)/usr) \
-	--with-libz-prefix=$(STAGING_DIR)/usr
+CMAKE_OPTIONS += \
+	-DBUILD_SHARED_LIBS=ON \
+	-DENABLE_ZLIB_COMPRESSION=ON \
+	-DCLEAR_MEMORY=ON
+
+ifeq ($(CONFIG_LIBSSH2_OPENSSL),y)
+	CMAKE_OPTIONS += -DCRYPTO_BACKEND=OpenSSL
+endif
```
Comment (neheb) [inline libs/libssh2/Makefile]: Could be rewritten as else. Otherwise just style.

**Comment by BKPepe** [comment]
Was it run-tested?

**Comment by jslachta** [comment]
Looks good to me. I'll do some run tests.

**Comment by ptpt52** [comment]
@jslachta @diizzyy @neheb Please fix the issue
I compile it on Ubuntu 18.04 and failed

**Pull request merged**